sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs

```diff
@@ -20,771 +21,1256 @@ namespace Azure.Messaging.ServiceBus.Diagnostics
-        public void SendMessageStart(string identifier, int messageCount)
+        // parameterless constructor for mocking
+        internal ServiceBusEventSource() { }
+
+        #region event constants
+        // event constants should not be changed
+        internal const int SendMessageStartEvent = 1;
+        internal const int SendMessageCompleteEvent = 2;
+        internal const int SendMessageExceptionEvent = 3;
+
+        internal const int CreateMessageBatchStartEvent = 4;
+        internal const int CreateMessageBatchCompleteEvent = 5;
+        internal const int CreateMessageBatchExceptionEvent = 6;
+
+        internal const int ReceiveMessageStartEvent = 7;
+        internal const int ReceiveMessageCompleteEvent = 8;
+        internal const int ReceiveMessageExceptionEvent = 9;
+
+        internal const int ScheduleMessageStartEvent = 10;
+        internal const int ScheduleMessageCompleteEvent = 11;
+        internal const int ScheduleMessageExceptionEvent = 12;
+
+        internal const int CancelScheduledMessageStartEvent = 13;
+        internal const int CancelScheduledMessageCompleteEvent = 14;
+        internal const int CancelScheduledMessageExceptionEvent = 15;
+
+        internal const int CompleteMessageStartEvent = 16;
+        internal const int CompleteMessageCompleteEvent = 17;
+        internal const int CompleteMessageExceptionEvent = 18;
+
+        internal const int DeferMessageStartEvent = 19;
+        internal const int DeferMessageCompleteEvent = 20;
+        internal const int DeferMessageExceptionEvent = 21;
+
+        internal const int AbandonMessageStartEvent = 22;
+        internal const int AbandonMessageCompleteEvent = 23;
+        internal const int AbandonMessageExceptionEvent = 24;
+
+        internal const int DeadLetterMessageStartEvent = 25;
+        internal const int DeadLetterMessageCompleteEvent = 26;
+        internal const int DeadLetterMessageExceptionEvent = 27;
+
+        internal const int PeekMessageStartEvent = 28;
+        internal const int PeekMessageCompleteEvent = 29;
+        internal const int PeekMessageExceptionEvent = 30;
+
+        internal const int RenewMessageLockStartEvent = 31;
+        internal const int RenewMessageLockCompleteEvent = 32;
+        internal const int RenewMessageLockExceptionEvent = 33;
+
+        internal const int ReceiveDeferredMessageStartEvent = 34;
+        internal const int ReceiveDeferredMessageCompleteEvent = 35;
+        internal const int ReceiveDeferredMessageExceptionEvent = 36;
+
+        internal const int LinkStateLostEvent = 37;
+        internal const int SessionReceiverLinkClosedEvent = 38;
+        internal const int AmqpLinkRefreshStartEvent = 39;
+        internal const int AmqpLinkRefreshCompleteEvent = 40;
+        internal const int AmqpLinkRefreshExceptionEvent = 41;
+        internal const int RunOperationExceptionEvent = 43;
```

Comment:
42 is missing :smile:

Reply:
Hah yeah, I suppose I can just fill it in with the latest event.